nrf_modem/doc/sockets/socket_options_func.rst

```diff
@@ -152,6 +154,33 @@ NRF_SO_EXCEPTIONAL_DATA
 
    The socket option is supported from modem firmware v2.0.0.
 
+NRF_SO_KEEPOPEN
+   Keep the socket from forcibly being closed upon PDN disconnection or reactivation events, or when the device is put into flight mode.
+   Until the PDN connection is reestablished, the socket is not functional and socket operations will return errors as appropriate (in general, ``NRF_ENETUNREACH``).
+   Upon PDN connection reestablishment, the socket behavior depends on the socket type and protocol, and on the IP address of the socket's newly established PDN connection, as shown in the table below.
+
+   +-----------------+--------------------+--------------------------------------------------------------+
+   | Socket type     | Socket protocol    | Socket is functional (no errors)                             |
+   +=================+====================+==============================================================+
+   | NRF_SOCK_DGRAM  | NRF_IPPROTO_UDP    | Always                                                       |
+   +-----------------+--------------------+--------------------------------------------------------------+
+   | NRF_SOCK_DGRAM  | NRF_SPROTO_DTLS1v2 | If peer supports DTLS connection ID                          |
+   +-----------------+--------------------+--------------------------------------------------------------+
+   | NRF_SOCK_STREAM | NRF_IPPROTO_TCP    | If the IP address of socket's PDN connection has not changed |
+   +-----------------+--------------------+--------------------------------------------------------------+
+   | NRF_SOCK_STREAM | NRF_SPROTO_TLS1v2  | If the IP address of socket's PDN connection has not changed |
+   +-----------------+--------------------+--------------------------------------------------------------+
+   | NRF_SOCK_RAW    | Any                | Always                                                       |
+   +-----------------+--------------------+--------------------------------------------------------------+
+
+   If the conditions to keep the socket open after PDN connection reestablishment are not met, the socket will report an error (in general, ``NRF_ENETDOWN``), and must be closed by the application.
+   For further information on how to detect and handle these errors, see :ref:`handling_pdn_errors_on_sockets`.
+
+.. note::
+   Putting the device into functional mode ``0`` (``+CFUN=0``) always forces all sockets to be closed, regardless of the :c:macro:`NRF_SO_KEEPOPEN` socket option.
+
+   This socket option is supported from modem firmware v2.0.1.
```

Comment:
This could be added to the public doc once the modem firmware v2.0.1 is released?